heareval/task_embeddings.py

```diff
@@ -22,102 +22,117 @@
     https://github.com/neuralaudio/hear2021-eval-kit/issues/49
 """
 
-import csv
 import glob
+import json
 import os.path
 from importlib import import_module
-from typing import Any, Dict, Tuple
+from typing import Any, Dict, Optional, Tuple
 
 import numpy as np
+import pandas as pd
 import soundfile as sf
 import torch
 from torch.utils.data import DataLoader, Dataset
 from tqdm.auto import tqdm
 
 # This could instead be something from the participants
-EMBEDDING_PIP = "heareval.model.baseline"
+# TODO: Command line or config?
+EMBEDDING_PIP = "hearbaseline"
 EMBEDDING_MODEL_PATH = ""  # Baseline doesn't load model
+# TODO: Actually load baseline model
 
 # TODO: Support for multiple GPUs?
 device = "cuda" if torch.cuda.is_available() else "cpu"
 
 EMBED = import_module(EMBEDDING_PIP)
 
 
-class CSVDataset(Dataset):
+class AudioFileDataset(Dataset):
     """
-    Read in a CSV file, and return data rows as [column 1, (column 2, ...)]
+    Read in a CSV file, and return audio filenames.
     """
 
     def __init__(self, csv_file):
-        # Our CSV files don't have headers, so we can't use
-        # pd.read_csv
-        # I'm on the fence whether we want CSV headers or not,
-        # since the format is standardized.
-        with open(csv_file) as f:
-            csvreader = csv.reader(f)
-            self.rows = [row for row in csvreader]
-        ncol = len(self.rows[0])
-        # Make sure all rows have the same number of column,
-        # and rewrite as [col1, (col2, ...)]
-        for idx, row in enumerate(self.rows):
-            assert len(row) == ncol
-            self.rows[idx] = [row[0], row[1:]]
+        self.rows = pd.read_csv(csv_file)
+        # Since event detection metadata will have duplicates, we de-dup
+        # TODO: This suggests we might want to, instead of using CSV files,
+        # have a JSONL for metadata, with one file per line.
+        # Then there can be timestamp: list[labels].
+        # TODO: Make this a generic util function somewhere?
+        # TODO: Is this the right column to dedup on ???
```

Comment:
Why not, instead of use the task metadata, just look through all the audio files in the task directory and compute embeddings for them? Then we don't need to worry about all this reading of metadata and de-dupping stuff. We ultimately want embeddings for all the audio files so let's just use them and save the embeddings in the same folder structure with the same filename as the audios?